.github/workflows/ci.yml

```diff
@@ -7,21 +7,49 @@ on:
     branches: [ main ]
 
 jobs:
-  build:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    permissions: {}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        submodules: false  # Currently not necessary
+
+    - name: Set up Node
+      uses: actions/setup-node@v3
+      with:
+        node-version: 18
 
+    - name: Install JSON validator
+      run: npm install -g ajv-cli@5 ajv-formats@2
+
+    - name: Validate features table
+      run: ajv -c ajv-formats -s features.schema.json -d features.json --strict=true --errors=text
+
+  build:
+    name: Build & Deploy
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
```

Comment:
Why is write permission necessary now?

Reply:
https://github.com/peaceiris/actions-gh-pages/commit/414c92c5ff594b380dd8df46dffbb34e3dd2343f

Reply:
If it's the default, then why is this change necessary?

Reply:
It is not necessary. I did this for… symmetry I guess, because I've also set `permissions: {}` for the test job. And I did _that_, because I'm executing arbitrary code from the Internet (`npm install ajv`) so limiting the permission removes the (albeit extremely minuscule) risk of supply chain attacks. There're no drawbacks, and something something the principle of least privilege, so I thought why not.